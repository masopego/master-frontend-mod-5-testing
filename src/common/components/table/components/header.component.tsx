import React from 'react';
import { HeaderGroup } from 'react-table';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import { CellComponent } from './cell.component';

interface Props {
  headerGroups: HeaderGroup[];
}

export const HeaderComponent: React.FunctionComponent<Props> = (props) => {
  const { headerGroups } = props;
  return (
    <TableHead>
      {headerGroups.map((headerGroup, index) => (
        <TableRow {...headerGroup.getHeaderGroupProps()} key={index}>
          {headerGroup.headers.map((column, index) => (
            <CellComponent {...column.getHeaderProps()} key={index}>
              {column.render('Header')}
            </CellComponent>
          ))}
        </TableRow>
      ))}
    </TableHead>
  );
};
