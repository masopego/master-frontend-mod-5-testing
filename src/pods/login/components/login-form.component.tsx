import React from 'react';
import { Formik, Form } from 'formik';
import Button from '@material-ui/core/Button';
import { TextFieldComponent } from 'common/components';
import { Login, createEmptyLogin } from '../login.vm';
import { formValidation } from '../login.validation';
import * as classes from './login-form.styles';
import { literals } from 'core/i18n';

interface Props {
  onLogin: (login: Login) => void;
}

export const LoginFormComponent: React.FunctionComponent<Props> = (props) => {
  const { onLogin } = props;
  return (
    <Formik
      onSubmit={onLogin}
      initialValues={createEmptyLogin()}
      validate={formValidation.validateForm}
    >
      {() => (
        <Form className={classes.root}>
          <TextFieldComponent
            name="user"
            label={`${literals.components.fields.user} *`}
            variant="outlined"
            data-cy="username-input"
          />
          <TextFieldComponent
            name="password"
            label={`${literals.components.fields.password} *`}
            type="password"
            variant="outlined"
            data-cy="password-input"
          />
          <Button
            type="submit"
            variant="contained"
            color="primary"
            className={classes.submit}
            data-cy="submit-button"
          >
            Login
          </Button>
        </Form>
      )}
    </Formik>
  );
};
