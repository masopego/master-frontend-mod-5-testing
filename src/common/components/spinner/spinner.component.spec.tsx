import React from 'react';
import {
  render,
  screen,
  act,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import { trackPromise } from 'react-promise-tracker';
import { SpinnerComponent } from 'common/components';

const timeout = (milliseconds = 500) =>
  new Promise(function (resolve, reject) {
    setTimeout(() => {
      resolve('done');
      reject(new Error('Whoops!'));
    }, milliseconds);
  });

describe('common/spinnerComponent', () => {
  it('should not be displayed spinner component if there is not any promise', () => {
    render(<SpinnerComponent />);

    const spinner = screen.queryByRole('presentation');

    expect(spinner).not.toBeInTheDocument();
  });

  it('should be displayed spinner component if there is a promise', () => {
    render(<SpinnerComponent />);

    act(() => {
      trackPromise(timeout());
    });

    const spinner = screen.getByRole('presentation');

    expect(spinner).toBeVisible();
  });

  it('render component should check with Snapshot', () => {
    render(<SpinnerComponent />);

    act(() => {
      trackPromise(timeout());
    });

    const spinner = screen.getByRole('presentation');

    expect(spinner).toMatchSnapshot();
  });

  it('should remove spinner component after promise is resolved', async () => {
    render(<SpinnerComponent />);

    act(() => {
      trackPromise(timeout());
    });

    const spinner = screen.getByRole('presentation');
    expect(spinner).toBeInTheDocument();
    expect(spinner).toBeVisible();

    await waitForElementToBeRemoved(() => screen.queryByRole('presentation'));
  });
});
