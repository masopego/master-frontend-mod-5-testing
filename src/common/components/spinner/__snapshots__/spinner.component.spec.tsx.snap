// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/spinnerComponent render component should check with Snapshot 1`] = `
<div
  class="css-1hppjzv"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="css-1gb2dig"
    tabindex="-1"
  >
    <span
      class="css-1xdhyk6"
    >
      <span
        class="css-129peub"
      />
      <span
        class="css-prnvxp"
      />
      <span
        class="css-8el4s5"
      />
      <span
        class="css-1hf2v34"
      />
      <span
        class="css-tx0a7l"
      />
    </span>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
